main.cpp

```diff
@@ -52,57 +49,47 @@ struct U
     float uVariable1 { 0.0f }, uVariable2 { 0.0f };
     //float* ptrTouVariable1 = &uVariable1;
     //float* ptrTouVariable2 = &uVariable2;
-    float updateU(float* updatedValue)      //12
+    float updateU(float& updatedValue)      //12
     {
-        if( updatedValue != nullptr)
+        std::cout << "U's unit1 value: " << this->uVariable1 << std::endl;
+        this->uVariable1 = updatedValue;
+        std::cout << "U's unit1 updated value: " << this->uVariable1 << std::endl;
+        while( std::abs(this->uVariable2 - this->uVariable1) > 0.001f )
         {
-            std::cout << "U's unit1 value: " << this->uVariable1 << std::endl;
-            this->uVariable1 = *updatedValue;
-            std::cout << "U's unit1 updated value: " << this->uVariable1 << std::endl;
-            while( std::abs(this->uVariable2 - this->uVariable1) > 0.001f )
-            {
-                if(this->uVariable2 < this->uVariable1)
-                    this->uVariable2 += 0.1f;
-                else
-                    this->uVariable2 -= 0.1f;
-            }
-            std::cout << "U's unit2 updated value: " << this->uVariable2 << std::endl;
-            return this->uVariable2 * this->uVariable1;
+            if(this->uVariable2 < this->uVariable1)
+                this->uVariable2 += 0.1f;
+            else
+                this->uVariable2 -= 0.1f;
         }
-        std::cout << "Warning: float updatedValue is nullptr!" << std::endl;
-        return 0.0f;
+        std::cout << "U's unit2 updated value: " << this->uVariable2 << std::endl;
+        return this->uVariable2 * this->uVariable1;
     }
     
 };
 
 struct multiplyUVariables
 {
-    static float multiply(U* uInstance, float* newValue)         //10
-    {
-        if(uInstance == nullptr || newValue == nullptr)
-        {
-            std::cout << "Warning: function arguments are nullptr" << std::endl;
-            return 0.0f;
-        }
-        std::cout << "U's uVariable1 value: " << uInstance->uVariable1 << std::endl;
-        uInstance->uVariable1 = *newValue;
-        std::cout << "U's uVariable1 updated value: " << uInstance->uVariable1 << std::endl;
+    static float multiply(U& uInstance, float* newValue)         //10
+    {            
+        std::cout << "U's uVariable1 value: " << uInstance.uVariable1 << std::endl;
+        uInstance.uVariable1 = *newValue;
+        std::cout << "U's uVariable1 updated value: " << uInstance.uVariable1 << std::endl;
```

Comment:
You can convert `newValue` to a reference in the function declaration
- [ ] revise